provider/openstack/provider.go

```diff
@@ -1233,6 +1233,12 @@ func (e *Environ) startInstance(
 			var serverDetail *nova.ServerDetail
 			serverDetail, err = waitForActiveServerDetails(client, server.Id, 5*time.Minute)
 			if err != nil || serverDetail == nil {
+				// If we got an error back (eg. StillBuilding)
+				// we need to terminate the instance before
+				// retrying to avoid leaking resources.
+				if termErr := e.terminateInstances(ctx, []instance.Id{instance.Id(server.Id)}); termErr != nil {
```

Comment:
Suggest adding an Info to day we're deleting the instance, like ln 1255.

Reply:
Do any of the provider's handle that case? 

The instance is leaked, as the instance id isn't saved until after it leaves the provider.